modules/conversantAnalyticsAdapter.js

```diff
@@ -77,33 +87,94 @@ export const cnvrHelper = {
+              onAdRenderFailed(args);
+              break;
+            case BID_WON:
+              onBidWon(args);
+              break;
+            case BID_TIMEOUT:
+              onBidTimeout(args);
+              break;
+            case BIDDER_ERROR:
+              onBidderError(args)
+          } // END switch
+        } else {
+          logMessage(CNVR_CONSTANTS.LOG_PREFIX + ' - ' + eventType + ': skipped due to sampling');
+        }// END IF(cnvrHelper.doSample)
+      } catch (e) {
+        // e = {stack:"...",message:"..."}
+        logError(CNVR_CONSTANTS.LOG_PREFIX + 'Caught error in handling ' + eventType + ' event: ' + e.message);
+        cnvrHelper.sendErrorData(eventType, e);
+      }
     } // END track()
   }
 );
 
 // ================================================== EVENT HANDLERS ===================================================
+
+/**
+ * Handler for BIDDER_ERROR events, tries to capture as much data, save it in cache which is then picked up by
+ * AUCTION_END event and included in that payload. Was not able to see an easy way to get adUnitCode in this event
+ * so not including it for now.
+ * https://docs.prebid.org/dev-docs/bidder-adaptor.html#registering-on-bidder-error
+ * Trigger when the HTTP response status code is not between 200-299 and not equal to 304.
+ {
+    error: XMLHttpRequest,  https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest
+    bidderRequest: {  https://docs.prebid.org/dev-docs/bidder-adaptor.html#registering-on-bidder-error
+        {
+            auctionId: "b06c5141-fe8f-4cdf-9d7d-54415490a917",
+            auctionStart: 1579746300522,
+            bidderCode: "myBidderCode",
+            bidderRequestId: "15246a574e859f",
+            bids: [{...}],
+            gdprConsent: {consentString: "BOtmiBKOtmiBKABABAENAFAAAAACeAAA", vendorData: {...}, gdprApplies: true},
+            refererInfo: {
+                canonicalUrl: null,
+                page: "http://mypage.org?pbjs_debug=true",
+                domain: "mypage.org",
+                ref: null,
+                numIframes: 0,
+                reachedTop: true,
+                isAmp: false,
+                stack: ["http://mypage.org?pbjs_debug=true"]
+            }
+        }
+    }
+}
+ */
+function onBidderError(args) {
+  let error = args.error;
+  let bidRequest = args.bidderRequest;
+  let auctionId = bidRequest.auctionId;
+  let bidderCode = bidRequest.bidderCode;
+  logWarn('onBidderError(): error received from bidder ' + bidderCode + '. Status = ' + error.status + ', Message = ' + error.statusText);
+  let errorObj = {
+    status: error.status,
+    message: error.statusText,
+    bidderCode: bidderCode,
+    url: cnvrHelper.getPageUrl(),
+  };
+  if (cnvrHelper.bidderErrorCache[auctionId]) {
+    cnvrHelper.bidderErrorCache[auctionId]['errors'].push(errorObj);
+  } else {
+    cnvrHelper.bidderErrorCache[auctionId] = {
+      errors: [errorObj],
+      timeReceived: Date.now()
```

Comment:
Only the first error has a timestamp?

Reply:
@pycnvr yes since they are relative to an auction, I want to grab them all as a list.  The timeReceived is there to purge the cache in case it doesn't get removed by an AUCTION_END event.